.github/workflows/on-pullrequest.yml

```diff
@@ -50,6 +50,8 @@ jobs:
   code-analysis:
     name: Code Analysis
     runs-on: ubuntu-latest
+    permissions:
+        security-events: write
```

Comment:
For some unknown reason CodeQL was working on a pull request, but it fails in `main` because it does not have write permissions.